solr/solr-bridge/src/main/java/org/fao/geonet/solr/SolrHTTPProxy.java

```diff
@@ -0,0 +1,386 @@
+                streamToClient = new GZIPOutputStream(response.getOutputStream());
+            } else if ("deflate".equalsIgnoreCase(contentEncoding) && !isCharsetKnown) {
+                // same but with deflate
+                streamFromServer = new DeflaterInputStream(connectionWithFinalHost.getInputStream());
+                streamToClient = new DeflaterOutputStream(response.getOutputStream());
+            } else {
+                throw new UnsupportedOperationException("Please handle the stream when it is encoded in " + contentEncoding);
+            }
+
+            byte[] buf = new byte[1024]; // read maximum 1024 bytes
+            int len;                     // number of bytes read from the stream
+            boolean first = true;        // helps to find the encoding once and only once
+            String s = "";               // piece of file that should contain the encoding
+            while ((len = streamFromServer.read(buf)) > 0) {
+
+                if (first && !isCharsetKnown) {
+                    // charset is unknown try to find it in the file content
+                    for (int i = 0; i < len; i++) {
+                        s += (char) buf[i]; // get the beginning of the file as ASCII
+                    }
+
+                    // s has to be long enough to contain the encoding
+                    if (s.length() > 200) {
+                        String charset = getCharset(s); // extract charset
+
+                        if (charset == null) {
+                            // the charset cannot be found, IE users must be warned
+                            // that the request cannot be fulfilled, nothing good would happen otherwise
+                            if (request.getHeader("User-Agent").toLowerCase().contains("msie")) {
+                                response.sendError(HttpServletResponse.SC_NOT_ACCEPTABLE,
+                                        "Charset of the response is unknown");
+                                streamFromServer.close();
+                                connectionWithFinalHost.disconnect();
+                                streamToClient.close();
+                                return;
+                            }
+                        } else {
+                            // override content-type header and add the charset found
+                            response.addHeader("Content-Type",
+                                    connectionWithFinalHost.getContentType()
+                                            + ";charset=" + charset);
+                            first = false; // we found the encoding, don't try to do it again
+                        }
+                    }
+                }
+
+                // for everyone, the stream is just forwarded to the client
+                streamToClient.write(buf, 0, len);
+            }
+
+            streamFromServer.close();
```

Comment:
SolrHTTPProxy was improved a lot in the Solr branch. I would let it like it is for now in this one.